### PR TITLE
Enable scaling grouping with scalar variables

### LINKED ISSUE #22042: Cannot include scalar variables in `scaling_group_variables` parameter
## Bug Description
If a scalar variable is included in the `Executioner/scaling_group_variables` parameter, then the user receives a cryptic error message

## Steps to Reproduce
Include a scalar variable in the `Executioner/scaling_group_variables` parameter

## Impact
Confusing experience for users trying to use automatic scaling and `scaling_group_variables`

### PR DESCRIPTION
**Pull request by lindsayad**
This refactors scaling code to be agnostic of whether a variable
is a field variable or a scalar variable

Closes #22042

### CODE REVIEW

**Review by GiudGiud — DISMISSED** [review]

**Comment by lindsayad** [comment]
this should be ready for review again

File: framework/include/systems/NonlinearSystemBase.h [hunk 1]
```diff
@@ -955,11 +955,8 @@ class NonlinearSystemBase : public SystemBase, public PerfGraphInterface
   /// like for solid/fluid mechanics
   std::vector<std::vector<std::string>> _scaling_group_variables;
 
-  /// Container to hold flag if field variable is to partipate in autoscaling
-  std::vector<bool> _field_variable_autoscaled;
-
-  /// Container to hold flag if scalar variable is to partipate in autoscaling
-  std::vector<bool> _scalar_variable_autoscaled;
+  /// Container to hold flag if variable is to partipate in autoscaling
```
Comment (GiudGiud) [inline framework/include/systems/NonlinearSystemBase.h:958]:
```suggestion
  /// Container to hold flag if variable is to participate in autoscaling
```

File: framework/src/systems/NonlinearSystemBase.C [hunk 2]
```diff
@@ -3484,34 +3484,42 @@ NonlinearSystemBase::setupScalingData()
...
 
   if (_scaling_group_variables.empty())
   {
-    _var_to_group_var.reserve(field_variables.size());
-    _num_scaling_groups = field_variables.size();
+    _var_to_group_var.reserve(n_vars);
+    _num_scaling_groups = n_vars;
 
-    for (auto & field_var : field_variables)
-      _var_to_group_var.insert(std::make_pair(field_var->number(), field_var->number()));
```
Comment (GiudGiud) [inline framework/src/systems/NonlinearSystemBase.C:3495]: any other variable type beyond field and scalar I would be unaware of?
this code was all to skip scalar variables

**Review by GiudGiud — CHANGES_REQUESTED** [review]
needs a test of using a scalar variable grouped in with a field variable.

FV diffusion + a flux + a lagrange multiplier to keep the average could be cool

File: framework/src/systems/NonlinearSystemBase.C (hunk 2 above)
Comment (lindsayad) [reply framework/src/systems/NonlinearSystemBase.C:3495]: nopers

Comment (lindsayad) [reply framework/src/systems/NonlinearSystemBase.C:3495]: https://mooseframework.inl.gov/docs/doxygen/moose/classMooseVariableBase.html

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**